prototype_android/src/main/java/com/myworkout/prototype/android/modules/authentication/LoginViewModel.kt

```diff
@@ -0,0 +1,32 @@
+package com.myworkout.prototype.android.modules.authentication
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.myworkout.shared.app.usecases.LoginUseCase
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+class LoginViewModel(
+    private val loginUseCase: LoginUseCase
+) : ViewModel() {
+
+    private val _isAuthorized = MutableLiveData<Boolean>()
+    val isAuthorized: LiveData<Boolean>
+        get() = _isAuthorized
+
+    val userName = MutableLiveData<String>()
+    val password = MutableLiveData<String>()
```

Comment:
I wonder how similar our ViewModels will be in the android vs iOS implementation. I see in the original implementation they use methods to pass data (e.g. `updateBreed(...)`). 
E.g. `userName` and `password` would not be settable in the same way using MutableLiveData in the iOS version so I think we in the LoginViewModel on iOS would need to use something like `updateUsername(username: String)`

Reply:
I think we shouldn't expect the native ViewModels to look very similar, except how they are integrating the UseCases. If I remember correctly one of the main reasons why we decided to go for native ViewModels was that we wanted to have flexibility on leveraging the strengths of each platform when it comes to hooking up with the views. This could be on iOS SwiftUI and on Android, since Jetpack Compose is not stable yet, Databinding.